rollup-node.md

```diff
@@ -0,0 +1,285 @@
+
+When included in the `transactions` array, this transaction should be
+RLP-encoded in the same way as other transactions.
+
+Here is an example valid `PayloadAttributesOPV1` object, which contains an L1
+attributes transaction as well as a single deposit:
+
+```js
+{
+  timestamp: "0x61a6336f",
+  random: "0xde5dff2b0982ecbbd38081eb8f4aed0525140dc1c1d56f995b4fa801a3f2649e",
+  suggestedFeeRecipient: "0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B",
+  transactions: [
+    "TODO specify L1 attribute transaction",
+    "0x02f87101058459682f0085199c82cc0082520894ab5801a7d398351b8be11c439e05c5b3259aec9b8609184e72a00080c080a0a6d217a91ea344fc09f740f104f764d71bb1ca9a8e159117d2d27091ea5fce91a04cf5add5f5b7d791a2c4663ab488cb581df800fe0910aa755099ba466b49fd69"
+  ]
+}
+```
+
+### Building the L2 block with the execution engine
+[calling-exec-engine]: #building-the-L2-block-with-the-execution-engine
+
+The Optimistic Ethereum [execution engine] is [specified in its own
+document][execution-engine-spec].
+
+This section defines how the rollup driver must interact with the execution
+engine's in order to convert [payload attributes] into L2 blocks.
+
+> **TODO** This section probably includes too much redundant details that will
+> need to be removed once the execution engine spec is up.
+
+Optimistic Ethereum's execution engine API is built upon [Ethereum's Engine API
+specification][eth-engine-api], with a couple of modifications. That
+specification builds upon [Ethereum's JSON-RPC API specification][JSON-RPC-API],
+which itself builds upon the [JSON-RPC specification][JSON-RPC].
+
+[eth-engine-api]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md
+[JSON-RPC]: https://www.jsonrpc.org/specification
+
+In particular, the [Ethereum's Engine API specification][eth-engine-api]
+specifies a [JSON-RPC] endpoint with a number of JSON-RPC (REST) routes, which
+are the means through which the rollup driver interacts with the execution
+engine.
+
+Instead of calling [`engine_forkchoiceUpdatedV1`], the rollup driver must call
+the new [`engine_forkchoiceUpdatedOPV1`] route. This has the same signature,
+except that:
+
+[`engine_forkchoiceUpdatedV1`]: https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#engine_forkchoiceupdatedv1
+[`engine_forkchoiceUpdatedOPV1`]: TODO
+
+> **TODO LINK** forkchoiceUpdatedOPV1 from spec
+
+- it takes a [`PayloadAttributesOPV1`] object as input instead of
+  [`PayloadAttributesV1`]. The execution engine must include the valid
+  transactions supplied in this object in the block, in the same order as they
+  were supplied, and only those. See the [previous section][payload-attr] for
+  the specification of how the properties must be set.
+
+- we repurpose the [`ForkchoiceStateV1`] structure with the following property
+  semantics:
+  - `headBlockHash`: block hash of the last block of the L2 chain, according to
+    the rollup driver.
+  - `safeBlockHash`: same as `headBlockHash`.
```

Comment:
This is changing on L1, and just a placeholder semantic, the two are not the same. And in the rollup they are definitely different. Also, this should be covered in the execution engine doc, not here in the general rollup-node doc.

Reply:
Yup, they're not the same but for the deposits-only stage we have nothing better to put in there, so I specified away.

I do think it belongs here too, because the rollup-node needs to fill in the arguments to make the call.

Reply:
I agree that what's currently written here is fine. It's a placeholder for when there is a sequencer and thus `unsafe` is more meaningful.